Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/SystemAllocator.h>
+
+#include <AzToolsFramework/FocusMode/FocusModeInterface.h>
+#include <AzToolsFramework/Prefab/PrefabFocusInterface.h>
+#include <AzToolsFramework/Prefab/Template/Template.h>
+
+namespace AzToolsFramework::Prefab
+{
+    class InstanceEntityMapperInterface;
+
+    //! Handles Prefab Focus mode, determining which prefab file entity changes will target.
+    class PrefabFocusHandler final
+        : private PrefabFocusInterface
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(PrefabFocusHandler, AZ::SystemAllocator, 0);
+
+        PrefabFocusHandler();
+        ~PrefabFocusHandler();
+
+        // PrefabFocusInterface override ...
+        PrefabFocusOperationResult FocusOnOwningPrefab(AZ::EntityId entityId) override;
+        TemplateId GetFocusedPrefabTemplateId() override;
+        InstanceOptionalReference GetFocusedPrefabInstance() override;
+        bool IsOwningPrefabBeingFocused(AZ::EntityId entityId) override;
+
+    private:
+        InstanceOptionalReference m_focusedInstance;
+        TemplateId m_focusedTemplateId;
```

Comment:
Since you are storing the instance ref and the templateId in this class, are you planning on sending out notifications from Instance and Template classes when they get destroyed so that this class can listen to them and clear out these values?
Or are you planning to just rely on the fact that FocusOnOwningPrefab() should be set correctly when things in the outliner get deleted?

Reply:
Hum, ideally it should not be possible to delete the instance that is being focused on, just like we don't allow deleting the root instance for the Outliner. There's no such check right now though.

I have taken a note and I will make sure that is taken care of in a follow-up PR (most likely the one that limits selection to non-focused entities since that's where it makes the most sense)